packages/store/src/internal/state-factory.ts

```diff
@@ -131,7 +131,8 @@ export class StateFactory {
           this.invokeActions(this._actions, action!).pipe(
             map(() => <ActionContext>{ action, status: ActionStatus.Successful }),
             defaultIfEmpty(<ActionContext>{ action, status: ActionStatus.Canceled }),
-            catchError(error => of(<ActionContext>{ action, status: ActionStatus.Errored, error }))
+            catchError(error => of(<ActionContext>{ action, status: ActionStatus.Errored, error })),
+            finalize(() => this._actions.next(<ActionContext>{ action, status: ActionStatus.Completed }))
```

Comment:
I don't think we should add a new lifecycle event, but rather filter for any of the existing ones.

Reply:
On the contrary, thanks to such processing, we solve the problem https://github.com/ngxs/store/issues/518

Reply:
@markwhitfeld Agreed.
@splincode the new `ofActionCompleted` should solve your given problem.